### PR TITLE
#139 the page about a single vale is missing at `/valves/[0-9]+`

### LINKED ISSUE #139: the page about a single vale is missing at `/valves/[0-9]+`
Let's add a page in `front_valves.rb` that will show all the details about a single valve, similar to how we show the information about a single job in `front_jobs.rb`. Let's also mention this page in `valves.enter()` when we notify a human via `human.notify()`. Now we mention just `https://www.zerocracy.com/valves`.

### PR DESCRIPTION
**Pull request by Suban05**
Closes #139

### CODE REVIEW

**Comment by Suban05** [comment]
@Yegorov please review

File: test/baza/test_valve.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    n = fake_name
+    b = fake_name
+    w = 'why'
+    human.valves.enter(n, b, w, job.id) { 42 }
+    valve = human.valves.get(human.valves.each.to_a.first[:id])
+    assert_equal(n, valve.name)
+    assert_equal(b, valve.badge)
+    assert_equal(w, valve.why)
+    assert_equal(job.id, valve.job.id)
+  end
```
Comment (Yegorov) [inline test/baza/test_valve.rb:51]: @Suban05 Maybe need to add check for result, e. g.:
```ruby
assert_equal("BAhpLw==\n", valve.result)
```

File: test/baza/test_valve.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    n = fake_name
+    b = fake_name
+    w = 'why'
+    human.valves.enter(n, b, w, nil) { 42 }
+    valve = human.valves.get(human.valves.each.to_a.first[:id])
+    assert_equal(n, valve.name)
+    assert_equal(b, valve.badge)
+    assert_equal(w, valve.why)
+    assert_nil(valve.job)
+  end
```
Comment (Yegorov) [inline test/baza/test_valve.rb:64]: @Suban05 Maybe need to add check for result, see above

File: test/front/test_front_valves.rb [hunk 3]
```diff
@@ -69,4 +69,13 @@ def test_valves
     get('/valves')
     assert_status(200)
   end
+
+  def test_read_valve
+    n = fake_name
+    fake_login(n)
+    human = app.humans.ensure(n)
+    post('/valves/add', 'name=hi&badge=abc&why=nothing')
+    get("/valves/#{human.valves.each.to_a.first[:id]}")
```
Comment (Yegorov) [inline test/front/test_front_valves.rb:78]: @Suban05 We really need call `.to_a` for Enumerator?
```ruby
get("/valves/#{human.valves.each.first[:id]}")
```
So will it work?

File: objects/baza/valve.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  end
+
+  def job
+    to_json[:job]
+  end
+
+  def to_json(*_args)
+    @to_json ||=
+      begin
+        row = @valves.pgsql.exec(
```
Comment (Yegorov) [inline objects/baza/valve.rb:70]: @Suban05 Maybe this line should look like this?
```ruby
row = pgsql.exec(
```

File: objects/baza/valve.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+          [@id, @valves.human.id]
+        ).first
+        raise Baza::Urror, "There is no valve ##{@id}" if row.nil?
+        {
+          id: @id,
+          name: row['name'].downcase,
+          created: Time.parse(row['created']),
+          badge: row['badge'],
+          why: row['why'],
+          result: row['result'],
```
Comment (Yegorov) [inline objects/baza/valve.rb:85]: @Suban05 Maybe we need to [decode](https://github.com/zerocracy/baza/blob/161da162116949ba5ac712302047e70c8517d37f/objects/baza/valves.rb#L70) result value?
Result value also used in [show page](https://github.com/zerocracy/baza/pull/257/files#diff-dce21a21e0a7bb937557f346c02009332bac86f54e3d274e993750ebcf918e2bR38)

File: test/baza/test_valve.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    assert_equal(job.id, valve.job.id)
+  end
+
+  def test_enter_valve_without_job
+    human = fake_human
+    n = fake_name
+    b = fake_name
+    w = 'why'
+    human.valves.enter(n, b, w, nil) { 42 }
+    valve = human.valves.get(human.valves.each.to_a.first[:id])
```
Comment (Yegorov) [inline test/baza/test_valve.rb:59]: @Suban05 We really need call `.to_a` for Enumerator?

File: test/baza/test_valve.rb [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    job = human.jobs.get(
+      human.tokens.add(fake_name).start(
+        fake_name, fake_name, 1, 0, 'n/a', ['hello, dude!', 'пока!'], '192.168.1.1'
+      ).id
+    )
+    n = fake_name
+    b = fake_name
+    w = 'why'
+    human.valves.enter(n, b, w, job.id) { 42 }
+    valve = human.valves.get(human.valves.each.to_a.first[:id])
```
Comment (Yegorov) [inline test/baza/test_valve.rb:46]: @Suban05 We really need call `.to_a` for Enumerator?

**Review by Yegorov — COMMENTED** [review]
@Suban05 Look good, but have some questions

**Comment by Suban05** [comment]
@Yegorov updated

**Review by Yegorov — APPROVED** [review]

**Comment by Suban05** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@Suban05 thanks!

**Comment by 0crat** [comment]
@Yegorov Thanks for the review! You've earned +4 points, which is the base reward for contributing. While it wasn't your own code (avoiding the -24 point deduction), the review didn't have enough comments to earn additional bonuses. Keep in mind that more detailed reviews with 6+ comments can boost your points next time. Your running balance is [+168](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). Keep up the good work! 👍

**Comment by 0crat** [comment]
@Suban05 Great job on your contribution! 🎉 You've earned +4 points, despite the high hits-of-code count. Your dedication is appreciated – [keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) up the good work! Your current balance stands at [+131](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). Looking forward to more of your valuable input!

**Pull request merged**